iothub/service/src/Authentication/IotHubConnectionProperties.cs

```diff
@@ -50,9 +50,14 @@ internal static string GetIotHubName(string hostName)
             }
 
             int index = hostName.IndexOf(IotHubConnectionStringConstants.HostNameSeparator, StringComparison.OrdinalIgnoreCase);
-            string iotHubName = index >= 0
-                ? hostName.Substring(0, index)
-                : hostName;
+
+            // throw if hostname is invalid format
+            if (index < 0)
+            {
+                throw new ArgumentException("Invalid hostname format. Hostname should be in the format of IOTHUB_NAME.azure-devices.net");
```

Comment:
FWIW, I think only Public Azure has a suffix of "azure-devices.net". We shouldn't reject all others.

```suggestion
                throw new ArgumentException("Invalid host name format. The value for 'HostName' should be in the format of IOTHUB_NAME.azure-devices.net");
```